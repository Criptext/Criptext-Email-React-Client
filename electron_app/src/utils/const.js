/* process */
require('dotenv').config();

const API_CLIENT_VERSION = '9.0.0';
const LINK_DEVICES_FILE_VERSION = '3';
const PROD_SOCKET_URL = 'wss://socket.criptext.com';
const PROD_SERVER_URL = 'https://api.criptext.com';
const PROD_DATA_TRANSFER_URL = 'https://transfer.criptext.com';
const FILE_SERVER_APP_ID = 'qynhtyzjrshazxqarkpy';
const FILE_SERVER_KEY = 'lofjksedbxuucdjjpnby';
const PROD_APP_DOMAIN = 'criptext.com';
const NEWS_SERVER_URL = 'https://news.criptext.com';

const isDevelopment = process.env.NODE_ENV === 'development';
if (isDevelopment) {
  console.log('\x1b[33m%s\x1b[0m', `\n\t Mode: ${process.env.NODE_ENV}\n`);
}

module.exports = {
  FILE_SERVER_APP_ID,
  FILE_SERVER_KEY,
  APP_DOMAIN: isDevelopment ? process.env.DEV_APP_DOMAIN : PROD_APP_DOMAIN,
  SERVER_URL: isDevelopment ? process.env.DEV_SERVER_URL : PROD_SERVER_URL,
  SOCKET_URL: isDevelopment ? process.env.DEV_SOCKET_URL : PROD_SOCKET_URL,
  DATA_TRANSFER_URL: isDevelopment
    ? process.env.DEV_DATA_TRANSFER_URL
    : PROD_DATA_TRANSFER_URL,
  NEWS_SERVER_URL,
  API_CLIENT_VERSION,
  LINK_DEVICES_FILE_VERSION
};
