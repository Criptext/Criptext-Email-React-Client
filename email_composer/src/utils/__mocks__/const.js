export const appDomain = 'criptext.com';

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  FORWARD: 'forward',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  NEW_WITH_DATA: 'new-with-data'
};

export const externalDomains = ['gmail.com', 'hotmail.com', 'yahoo.com'];
