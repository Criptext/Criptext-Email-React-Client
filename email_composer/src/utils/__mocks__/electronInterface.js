export const myAccount = {
  id: 1,
  recipientId: 'myAccountRecipientId',
  name: 'myAccountName'
};

export const emailKey = '1';
export const emailKeyWithFile = '2';

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  FORWARD: 'forward'
};

export const mySettings = {
  language: 'en',
  opened: 1,
  theme: 'light'
};
