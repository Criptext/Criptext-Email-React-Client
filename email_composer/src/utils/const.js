/*global process */
export const appDomain =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_APPDOMAIN
    : 'criptext.com';

export const avatarBaseUrl =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_AVATAR_URL
    : `https://api.criptext.com/user/avatar/`;

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  FORWARD: 'forward',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  NEW_WITH_DATA: 'new-with-data'
};

export const defaultEmptyMimetypeValue = 'application/octet-stream';

export const previewLength = 100;
