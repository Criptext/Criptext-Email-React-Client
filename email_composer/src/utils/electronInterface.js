import { labels } from './systemLabels';
const electron = window.require('electron');
const { remote, ipcRenderer } = electron;
const composerId = remote.getCurrentWindow().id;
const globalManager = remote.require('./src/globalManager');
export const getAlicePort = remote.require('./src/aliceManager').getPort;
export const getAlicePassword = remote.require('./src/aliceManager')
  .getPassword;

export const { FILE_SERVER_APP_ID, FILE_SERVER_KEY } = remote.require(
  './src/utils/const'
);

export const getEmailToEdit = () => {
  return globalManager.emailToEdit.get(composerId);
};

const accounts = remote.require('./src/Account');
export let myAccount = accounts.activeAccount;
export const setMyAccount = recipientId => {
  myAccount = accounts.find(account => account.recipientId === recipientId);
};
export const mySettings = remote.require('./src/Settings');
export const LabelType = labels;

/* Window events
   ----------------------------- */
export const sendEventToMailbox = (name, params) => {
  ipcRenderer.send(name, params);
};
