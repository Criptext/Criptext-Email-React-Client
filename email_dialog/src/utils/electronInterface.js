const electron = window.require('electron');
const { ipcRenderer, remote } = electron;
const globalManager = remote.require('./src/globalManager');

export const mySettings = remote.require('./src/Settings');
export const remoteData = globalManager.modalData.get();

/* Window events
   ----------------------------- */
export const onResponseDialog = (event, response, sendTo) => {
  ipcRenderer.send('response-dialog', response, sendTo);
};
