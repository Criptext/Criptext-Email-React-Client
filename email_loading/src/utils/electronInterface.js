import { labels } from './systemLabels';
const { remote } = window.require('electron');
const socketClient = remote.require('./src/socketClient');
const globalManager = remote.require('./src/globalManager');

export const myAccount = remote.require('./src/Account');

export const mySettings = remote.require('./src/Settings');
export const getAlicePort = remote.require('./src/aliceManager').getPort;

export const LabelType = labels;

export const { loadingType, remoteData, shouldResetPIN } = remote.getGlobal(
  'loadingData'
);

export const getMailboxGettingEventsStatus = () => {
  return globalManager.isGettingEvents.get();
};

export const setRemoteData = data => {
  globalManager.loadingData.set(data);
};

export const startSocket = jwt => {
  socketClient.start(jwt);
};

export const stopSocket = () => {
  return socketClient.disconnect();
};

export const isFromStore =
  globalManager.isWindowsStore.get() || globalManager.isMAS.get();

export const setPendingRestoreStatus = status => {
  globalManager.pendingRestore.set(status);
};
