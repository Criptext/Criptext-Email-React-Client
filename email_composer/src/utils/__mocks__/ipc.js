import { appDomain } from './const';
import file from './../../../public/emails.json';

const contacts = [
  {
    id: 1,
    name: 'Usuario 1',
    email: `usuario1@${appDomain}`
  },
  {
    id: 2,
    name: 'Usuario 2',
    email: `usuario2@${appDomain}`
  },
  {
    id: 3,
    name: 'Usuario 3',
    email: `usuario3@${appDomain}`
  },
  {
    id: 4,
    name: 'Usuario 4',
    email: `usuario4@${appDomain}`
  }
];

const emails = file.emails;

const files = [
  {
    id: 1,
    token: 'tokenFile1',
    name: 'Criptext_file_1.png',
    readOnly: 0,
    size: 149836,
    status: 1,
    date: '2018-11-23 17:24:36',
    mimeType: 'image/png',
    ephemeral: 0,
    ephemeralStart: 0,
    ephemeralTime: 0,
    emailId: emails[1].id,
    key: 'jocmpqwrwgonedbsfsq==',
    iv: 'cmwgnoveirgkddfg=='
  }
];

const fileKeys = [
  {
    id: 1,
    key: 'key_file1',
    iv: 'iv_file1',
    emailId: emails[1].id
  }
];

export const emailId = 1;

export const getContactsByEmailId = id => {
  const response = {
    to: [{ email: contacts[0].email }],
    cc: [{ email: contacts[1].email }],
    bcc: [{ email: contacts[2].email }],
    from: [{ email: contacts[3].email }]
  };
  const emptyResponse = {
    to: [],
    cc: [],
    bcc: [],
    from: [{ email: contacts[3].email }]
  };
  return id === emailId ? response : emptyResponse;
};

export const getEmailByKeyWithbody = key => {
  return emails.filter(email => email.key === key)[0];
};

export const getFilesByEmailId = emailId => {
  return files.filter(file => file.emailId === emailId);
};

export const getFileKeyByEmailId = emailId => {
  return fileKeys.filter(fileKey => fileKey.emailId === emailId);
};
