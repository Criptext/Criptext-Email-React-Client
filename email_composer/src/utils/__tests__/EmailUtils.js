/* eslint-env node, jest */

import {
  formDataToEditDraft,
  formDataToReply,
  formOutgoingEmailFromData
} from './../EmailUtils';
import { emailKey, emailKeyWithFile } from './../__mocks__/electronInterface';
import { appDomain } from './../const';

jest.mock('./../../utils/const');
jest.mock('./../../utils/ipc');
jest.mock('./../../utils/electronInterface');

describe('Edit draft: ', () => {
  it('Should load a draft correctly', async () => {
    const dataToEdit = await formDataToEditDraft(emailKey);
    dataToEdit.htmlBody = '<p>Modified body content</p>';
    expect(dataToEdit).toMatchSnapshot();
  });
});

describe('Reply, Reply-all, Forward: ', () => {
  it('Should load an email to reply', async () => {
    const dataToEdit = await formDataToReply(emailKey, 'reply');
    dataToEdit.htmlBody = '<p>Modified body content</p>';
    expect(dataToEdit).toMatchSnapshot();
  });

  it('Should load an email to reply to all', async () => {
    const dataToEdit = await formDataToReply(emailKey, 'reply-all');
    dataToEdit.htmlBody = '<p>Modified body content</p>';
    expect(dataToEdit).toMatchSnapshot();
  });

  it('Should load an email to forward', async () => {
    const dataToEdit = await formDataToReply(emailKey, 'forward');
    dataToEdit.htmlBody = '<p>Modified body content</p>';
    expect(dataToEdit).toMatchSnapshot();
  });

  it('Should load an email to forward: Load attachments', async () => {
    const dataToEdit = await formDataToReply(emailKeyWithFile, 'forward');
    dataToEdit.htmlBody = '<p>Modified body content</p>';
    expect(dataToEdit).toMatchSnapshot();
  });
});

describe('[Form outgoing email] ', () => {
  it('Should form outgoing email from data', () => {
    const labelId = 6;
    const composerData = {
      account: { id: 1 },
      toEmails: [`toUser@${appDomain}`],
      ccEmails: [`ccUser@${appDomain}`],
      bccEmails: [`bccUser@${appDomain}`],
      secure: true,
      textSubject: 'Subject',
      body: '<p>Hello</p>',
      files: [],
      labelId,
      status: 4
    };

    const outgoingData = formOutgoingEmailFromData(composerData);
    outgoingData.emailData.email.date = 1524861748481;
    outgoingData.emailData.email.key = 1524861748481;
    expect(outgoingData).toMatchSnapshot();
  });
});
