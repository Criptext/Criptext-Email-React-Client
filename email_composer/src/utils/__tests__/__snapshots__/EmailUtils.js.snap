// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit draft:  Should load a draft correctly 1`] = `
Object {
  "bccEmails": Array [
    Object {
      "complete": "<usuario3@criptext.com>",
      "email": "usuario3@criptext.com",
      "form": "tag-app-domain",
      "name": undefined,
    },
  ],
  "ccEmails": Array [
    Object {
      "complete": "<usuario2@criptext.com>",
      "email": "usuario2@criptext.com",
      "form": "tag-app-domain",
      "name": undefined,
    },
  ],
  "files": Array [],
  "htmlBody": "<p>Modified body content</p>",
  "textSubject": "Email subject 1",
  "threadId": "1",
  "toEmails": Array [
    Object {
      "complete": "<usuario1@criptext.com>",
      "email": "usuario1@criptext.com",
      "form": "tag-app-domain",
      "name": undefined,
    },
  ],
}
`;

exports[`Form outgoing email:  Should form outgoing email from data 1`] = `
Object {
  "bodyWithSign": "<p>Hello</p>",
  "emailData": Object {
    "body": "<p>Hello</p>",
    "email": Object {
      "content": "",
      "date": 1524861748481,
      "fromAddress": "myAccountName <myAccountRecipientId@criptext.com>",
      "key": 1524861748481,
      "preview": "Hello",
      "secure": true,
      "status": 4,
      "subject": "Subject",
      "threadId": undefined,
      "unread": false,
    },
    "files": Array [],
    "labels": Array [
      6,
    ],
    "recipients": Object {
      "bcc": Array [
        "bccUser@criptext.com",
      ],
      "cc": Array [
        "ccUser@criptext.com",
      ],
      "from": Array [
        "myAccountRecipientId@criptext.com",
      ],
      "to": Array [
        "toUser@criptext.com",
      ],
    },
  },
  "recipientDomains": Array [
    Object {
      "domain": "criptext.com",
      "recipientId": "toUser",
      "type": "to",
      "username": "toUser",
    },
    Object {
      "domain": "criptext.com",
      "recipientId": "ccUser",
      "type": "cc",
      "username": "ccUser",
    },
    Object {
      "domain": "criptext.com",
      "recipientId": "bccUser",
      "type": "bcc",
      "username": "bccUser",
    },
  ],
}
`;

exports[`Reply, Reply-all, Forward:  Should load an email to forward 1`] = `
Object {
  "bccEmails": Array [],
  "ccEmails": Array [],
  "files": Array [],
  "htmlBody": "<p>Modified body content</p>",
  "status": 1,
  "textSubject": "FW: Email subject 1",
  "threadId": "1",
  "toEmails": Array [],
}
`;

exports[`Reply, Reply-all, Forward:  Should load an email to forward: Load attachments 1`] = `
Object {
  "bccEmails": Array [],
  "ccEmails": Array [],
  "files": Array [
    Object {
      "fileData": Object {
        "date": "2018-11-23 17:24:36",
        "emailId": 2,
        "ephemeral": 0,
        "ephemeralStart": 0,
        "ephemeralTime": 0,
        "id": 1,
        "iv": "cmwgnoveirgkddfg==",
        "key": "jocmpqwrwgonedbsfsq==",
        "mimeType": "image/png",
        "name": "Criptext_file_1.png",
        "readOnly": 0,
        "size": 149836,
        "status": 1,
        "token": "tokenFile1",
        "type": "image/png",
      },
      "iv": "cmwgnoveirgkddfg==",
      "key": "jocmpqwrwgonedbsfsq==",
      "mode": "uploaded",
      "percentage": 100,
      "shouldDuplicate": true,
      "token": "tokenFile1",
    },
  ],
  "htmlBody": "<p>Modified body content</p>",
  "status": 1,
  "textSubject": "FW: Email subject 2",
  "threadId": "2",
  "toEmails": Array [],
}
`;

exports[`Reply, Reply-all, Forward:  Should load an email to reply 1`] = `
Object {
  "bccEmails": Array [],
  "ccEmails": Array [],
  "files": Array [],
  "htmlBody": "<p>Modified body content</p>",
  "status": 2,
  "textSubject": "RE: Email subject 1",
  "threadId": "1",
  "toEmails": Array [
    Object {
      "complete": "From1 <from1@criptext.com>",
      "email": "from1@criptext.com",
      "form": "tag-app-domain",
      "name": "From1",
    },
  ],
}
`;

exports[`Reply, Reply-all, Forward:  Should load an email to reply to all 1`] = `
Object {
  "bccEmails": Array [],
  "ccEmails": Array [
    Object {
      "complete": "<usuario2@criptext.com>",
      "email": "usuario2@criptext.com",
      "form": "tag-app-domain",
      "name": undefined,
    },
  ],
  "files": Array [],
  "htmlBody": "<p>Modified body content</p>",
  "status": 2,
  "textSubject": "RE: Email subject 1",
  "threadId": "1",
  "toEmails": Array [
    Object {
      "complete": "From1 <from1@criptext.com>",
      "email": "from1@criptext.com",
      "form": "tag-app-domain",
      "name": "From1",
    },
    Object {
      "complete": "<usuario1@criptext.com>",
      "email": "usuario1@criptext.com",
      "form": "tag-app-domain",
      "name": undefined,
    },
  ],
}
`;
